eng/native/gen-buildsys.cmd

```diff
@@ -2,26 +2,38 @@
 rem
 rem This file invokes cmake and generates the build system for windows.
 
-setlocal
+set __argCount=0
+for %%x in (%*) do set /A __argCount+=1
 
-set argC=0
-for %%x in (%*) do Set /A argC+=1
+if %__argCount% lss 4 goto :USAGE
+if %1=="/?" goto :USAGE
 
-if %argC% lss 4 GOTO :USAGE
-if %1=="/?" GOTO :USAGE
-
-setlocal enabledelayedexpansion
+set __Os=%5
 set "__repoRoot=%~dp0..\.."
 :: normalize
 for %%i in ("%__repoRoot%") do set "__repoRoot=%%~fi"
 
+:: Set up the EMSDK environment before setlocal so that it propagates to the caller.
+if /i "%__Os%" == "browser" (
+    if "%EMSDK_PATH%" == "" (
+        if not exist "%__repoRoot%\src\mono\browser\emsdk" (
+            echo Error: Should set EMSDK_PATH environment variable pointing to emsdk root.
+            exit /B 1
+        )
+        set EMSDK_QUIET=1 && call "%__repoRoot%\src\mono\browser\emsdk\emsdk_env"
+    ) else (
+        set EMSDK_QUIET=1 && call "%EMSDK_PATH%\emsdk_env"
+    )
+)
+
```

Comment:
we are missing the `WASI_SDK_PATH` fix from linux version here

Reply:
Ehhm, the `WASI_SDK_PATH` logic here (a bit below in the script) should be the same as in the `gen-buildsys.sh`.